2st-semester/oop/TrieHomework/Trie/Trie.cs

```diff
@@ -0,0 +1,156 @@
+namespace Trie;
+
+/// <summary>
+/// Adds a word to the Trie.
+/// Trie data structure.
+/// </summary>
+public class Trie
+{
+    private readonly Node root = new();
+
+    /// <summary>
+    /// Gets the number of words in the Trie.
+    /// </summary>
+    public int Size => this.root.CountPrefixOf;
+
+    /// <summary>
+    /// Adds a word to the Trie.
+    /// </summary>
+    /// <param name="word">The word to be added to the Trie.</param>
+    /// <returns>
+    /// True if the word was successfully added to the Trie. <br/> False if the word already exists in the Trie.
+    /// </returns>
+    public bool Add(string word)
+    {
+        if (string.IsNullOrEmpty(word))
+        {
+            return false;
+        }
+
+        var node = this.Find(word);
+        if (node is not null && node.IsEndOfWord)
+        {
+            return false;
+        }
+
+        var current = this.root;
+        current.CountPrefixOf++;
+        foreach (var character in word)
+        {
+            if (!current.Children.TryGetValue(character, out Node? value))
+            {
+                value = new Node();
+                current.Children.Add(character, value);
+            }
+
+            current = value;
+            current.CountPrefixOf++;
+        }
+
+        current.IsEndOfWord = true;
+        return true;
+    }
+
+    /// <summary>
+    /// Checks if the trie contains the specified word.
+    /// </summary>
+    /// <param name="word">The word to search for in the trie.</param>
+    /// <returns>True if the word is found, otherwise false.</returns>
+    public bool Contains(string word)
+    {
+        if (string.IsNullOrEmpty(word))
+        {
+            return false;
+        }
+
+        var node = this.Find(word);
+
+        if (node is null)
+        {
+            return false;
+        }
+
+        if (node.IsEndOfWord)
+        {
+            return true;
+        }
+
+        return false;
```

Comment:
Можно было бы написать
```suggestion
        return (node is not null) && (node.IsEndOfWord);
```
Но это дело вкуса: можно сказать, что это выражение длинное и плохо читается.